### DISCUSSION TITLE
About OC4 Free decay test

### DISCUSSION THREAD

**Discussion by icnosvbovwb** [OpenFAST/openfast#1414] [Q&A]
Dear @jjonkman 
![image](https://user-images.githubusercontent.com/107483730/213394146-16163e89-a6e4-4949-b757-ebd938582d10.png)
![image](https://user-images.githubusercontent.com/107483730/213394479-663696c9-8672-4583-a46b-2ec79334da0c.png)
![image](https://user-images.githubusercontent.com/107483730/213393935-7ed6672d-288d-4048-a904-84fdf299d823.png)
"Numerical study on passive structural control of semi-submersible floating 
wind turbine considering non-collinear wind and waves".
I see the free attenuation test of OC4 from this article and want to reproduce its process, but my results are a little different from them. The conditions and degrees of freedom I set are as follows:
CompInflow=0，CompAero=0, WaveMod=0, GenDOF=False, RotSpeed=0, BlPitch(1)=BlPitch(2)=BlPitch(3)=90.
![image](https://user-images.githubusercontent.com/107483730/213395688-cdc54336-2f98-4377-85bd-cc3ef0aa19a9.png)
I don't know why there are different results. Please indicate my mistakes. The calculation time step is 0.025, so the horizontal ordinate is 600/0.025=24000. I wonder if the tower is placed on one of the float bowl. If so, how to set the tower on the middle column

**Comment by jjonkman** [comment]
Dear @icnosvbovwb,

I'm not familiar with the paper you are referring to, but from a quick Google search, I believe you are referring to: https://www.sciencedirect.com/science/article/pii/S0029801822020285, which does not include NREL authors.  I would suggest reaching out to the authors of that paper for questions related to it.

That said, the three plots you circle (`PtfmSway`, `PtfmRoll`, and `PtfmYaw`) all have very small y-axis ranges, probably because your case is is mostly two-dimensional (with dominant motions only in the X-Z plane).

Best regards,

**Comment by icnosvbovwb** [comment]
Dear @jjonkman 
![image](https://user-images.githubusercontent.com/107483730/213865963-b6073173-6a8a-4961-ae4f-f7d4798c8543.png)
![image](https://user-images.githubusercontent.com/107483730/213866073-d41b3836-72e5-47be-9a6c-17ca774e9131.png)
How to set the tower at the middle column？

**Reply by jjonkman** [reply]
Dear @icnosvbovwb,

The properties of the platform in OpenFAST (structural properties in ElastoDyn or SubDyn, hydrodynamic properties in HydroDyn, and mooring properties in MoorDyn, MAP++, or FEAMooring) are defined in a body-fixed coordinate system that, when undisplaced, is aligned with the tower centerline intersecting the mean sea level.

Best regards,

**Comment by icnosvbovwb** [comment]
Dear @jjonkman 
I'm not sure how to set the Tower in the middle column ,rather than on the one of three pontoons. Just like this picture.
![image](https://user-images.githubusercontent.com/107483730/220843699-93189713-55f2-46a4-920b-093444d77a0a.png)

**Reply by jjonkman** [reply]
Dear @icnosvbovwb,

It looks like you are trying to model the OC4-DeepCwind semisubmersible.  There is already an OpenFAST model of this system available in the OpenFAST r-test--see: https://github.com/OpenFAST/r-test/tree/main/glue-codes/openfast/5MW_OC4Semi_WSt_WavesWN.

Best regards,

**Comment by icnosvbovwb** [comment]
Dear @jjonkman
1.The platform display of different types of wind turbines, such as using VTK to display Spar, OC4, TLP, etc., do we need to modify some values of PLATFORM ADDITIONAL STIFFNESS AND DAMPING to display? Because if I do not change by default, only the part above the tower will be displayed, and all platforms will not be displayed. 
![image](https://user-images.githubusercontent.com/107483730/221143747-7411a549-a434-49ef-8b24-08cee2f0cf51.png)
2.Another question is that what is the abbreviation of first order wave excitation force in OutListParameters?

3.Also, the TLP platform is invisible. I don't know where to set it
![image](https://user-images.githubusercontent.com/107483730/221196950-5aaec797-8121-4fc4-874d-0271cbe3846c.png)

**Reply by jjonkman** [reply]
Dear @icnosvbovwb.

I'm not sure I fully understand your question, but the additional platform stiffness and damping matrices will not effect the visualization.

Regarding why the floating platform is not displayed, see the following still unresolved OpenFAST issue: https://github.com/OpenFAST/openfast/issues/776.

Regarding your question about "first order wave excitation force", presumably you are referring to the potenial-flow solution.  The HydroDyn outputs for the first-order wave-excitation force for the 1st WAMT body are named `B1WvsF1xi`, `B1WvsF1yi`, and `B1WvsF1zi` in global coordinates (likewise for other WAMIT bodies, where the first 1 should become N for body N).

Best regards,

**Comment by icnosvbovwb** [comment]
Dear @jjonkman 
![image](https://user-images.githubusercontent.com/107483730/221197655-8568274e-13ac-45c2-a4a7-aadff9b36153.png)
This picture is the VTK of TLP, but why don't we see the platform? Please tell me how to set up to see the platform? I found that there was no MorisonSurface file generated. I think this may be the reason why the platform is not visible.

**Reply by jjonkman** [reply]
Dear @icnosvbovwb,

As I mentioned in my prior post, the floating platform cannot be visualized until issue #776 is resolved.

Best regards,

**Comment by axxyyyyyyyy378** [comment]
Dear @icnosvbovwb ,

I am a beginner of openfast, could you please tell me how to implement the initial conditions in the code when doing free-dacy？
![image](https://github.com/OpenFAST/openfast/assets/150985396/6757a490-727f-48a7-ab66-17ecb71700f8)

Best regards,

**Reply by jjonkman** [reply]
Dear @axxyyyyyyyy378,

Initial conditions for the six rigid-body platform degrees of freedom can be set within the ElastoDyn primary input.

Best regards,

**Reply by axxyyyyyyyy378** [reply]
Dear @jjonkman ,

Thanks for your reply, I have solved the problem, and got the result of free-deacy, but now I have the same problem as @icnosvbovwb， The initial conditions and degrees of freedom I set are the same as @icnosvbovwb ，such as CompInflow=0，CompAero=0, WaveMod=0, GenDOF=False, RotSpeed=0, BlPitch(1)=BlPitch(2)=BlPitch(3)=90. and the load conditions in free-decay simulations：
![image](https://github.com/OpenFAST/openfast/assets/150985396/915779e0-c6f1-4a2a-8890-111e7bab2f95)
As can be seen from the comparison of results below, the results for PtfmSurge, Ptfmpitch and PtfmHeave are exactly the same as in the paper，but the results for PtfmSway, PtfmRoll and PtfmYaw are bad, I'm confused, What do you think caused bad results?
the picture below is my result:
<img width="1406" alt="1700194397594" src="https://github.com/OpenFAST/openfast/assets/150985396/129bf6fe-ffd5-4963-809f-6fdcd5105aee">
the picture below is the result in paper:
![1700194651457](https://github.com/OpenFAST/openfast/assets/150985396/046cc411-8e17-4e63-9318-a7df98d7c08d)

Best regards,

**Reply by jjonkman** [reply]
Dear @axxyyyyyyyy378,

All of the plots you label as "bad" have very small very values, which have little (if any) coupling to the degrees of freedom initialized with free decay.  I'm guessing there are small imbalances that cause a small amount of coupling, but I wouldn't be concerned that these are different from some publication.

Best regards,

**Reply by axxyyyyyyyy378** [reply]
Dear @jjonkman ,
Very very thanks for your reply.

Best regards,

**Comment by axxyyyyyyyy378** [comment]
Dear @jjonkman ,

Sorry to bother, but I found another problem. My OpenFAST version is 3.5.1. In the case “5MW_OC4Semi_WSt_WavesWN” , I tried to reduce the water depth from 200m to 50m, and the length of the mooringline from 835.35 m to 735.35 m, all other conditions did not change, the strange thing is, the case was able to work with such a short length of chain（I was expecting an error）, in fact, 735.35 m is less than the distance between the anchor and the floating platform, but as the result picture below, the mooringline was just stretched, and it seems to work fine. My question is: Why is the mooringline so short, but it still works? Is it because it can be stretched like a spring?
![image](https://github.com/OpenFAST/openfast/assets/150985396/1967465b-e5d1-41dc-94b8-eb799f630b22)
And there is the vedio of my result:

https://github.com/OpenFAST/openfast/assets/150985396/b226d9b8-8b19-4247-afc6-8a932a8ef6ce


https://github.com/OpenFAST/openfast/assets/150985396/349a2e08-c48a-41f8-a512-7fbd57494cf4





Best regards,

**Reply by jjonkman** [reply]
Dear @axxyyyyyyyy378,

Yes, the mooring lines will act as springs if the lines are fully taut.

Best regards,

**Reply by axxyyyyyyyy378** [reply]
Dear @jjonkman,

I learned some basic theories about moordyn, and i know it uses a lumped-mass approach to discretize the cable dynamics over the length of the mooring line. The cable is broken up into n evenly-sized line segments connecting N + 1 node points. In my tests, I found that the number of nodes has an impact on OpenFAST's results, as follows:

<img width="615" alt="1700230458671" src="https://github.com/OpenFAST/openfast/assets/150985396/e83be11d-ad08-45e4-906b-fbfd22ed2368">

Without changing the other conditions, I only changed the number of nodes (NumSegs) , from 20 to 60, and then I output the six-degree-of-freedom motion curve of the FOWT  before and after below, I found that they were very different.
the result below is "NumSegs=20":
<img width="1145" alt="1700231366133" src="https://github.com/OpenFAST/openfast/assets/150985396/b8ca1655-56fa-41a7-83e2-aaa699cd2668">
<img width="1164" alt="1700231399940" src="https://github.com/OpenFAST/openfast/assets/150985396/013d330d-bbd1-4136-8859-fb19289326de">

the result below is "NumSegs=60":
<img width="1049" alt="1700231169605" src="https://github.com/OpenFAST/openfast/assets/150985396/e53e32ab-7be7-4a45-9c6a-0280654a0f59">
<img width="1061" alt="1700231145149" src="https://github.com/OpenFAST/openfast/assets/150985396/a3f5d917-c6a4-4396-9040-b07d61aa1faf">


So, Is the results before and after I change the number of nodes reasonable? If so, what causes the change? Look forward to your reply.

Best regards,

**Reply by axxyyyyyyyy378** [reply]
Dear @jjonkman,

Sorry to trouble you again, but I'd like a chance to ask all the questions. yes I have another question. Could you please tell me is the tension on the mooring line the same after it is dispersed into many nodes? I think because of the gravity it is different ,so is there a way to output the tension of each node? I don't see any other output parameters in the user manual, so I only know“FairTen” and“Anchten,” and I'm guessing“Fairten” refers to the tension at the node near the floating platform, “Anchten” is tension near an anchor on the sea floor, am i right? 
<img width="536" alt="1700232277836" src="https://github.com/OpenFAST/openfast/assets/150985396/b356699f-251e-4fe9-a68d-c42b10b91db7">

Best regards,

**Reply by jjonkman** [reply]
Dear @axxyyyyyyyy378,

It is always good to check the convergence of the solution as a function of discretization.  But the finer the discretization, the more computationally expensive the solution will be.  So, there is always a balancing game between accuracy and time-to-solution.

Yes, you can output the tension at each node.  See the MoorDyn documentation here for more information: https://moordyn.readthedocs.io/en/latest/inputs.html#outputs.

Best regards,

**Reply by axxyyyyyyyy378** [reply]
Dear @jjonkman ,

Thanks for your kind reply ,  I have learned a lot. 

Best regards,

**Reply by axxyyyyyyyy378** [reply]
Dear @jjonkman ,

In your previous answer, you mentioned “the mooring lines will act as springs if the lines are fully taut” , as we know，an important parameter of linear spring is stiffness coefficient, so in this case, what is the equivalent stiffness coefficient of mooring line?  I do not seem to find the relevant parameter in the moordyn input file, but I guess  "EA" divided by the total length of the mooring line equals the stiffness factor K , am i right？
![1700360556851](https://github.com/OpenFAST/openfast/assets/150985396/62d90298-4a66-42b8-94bf-9dfd4ee6e2a5)

And here's another problem. For OpenFAST's existing FOWT model, such as “5MW_OC4Semi_WSt_WavesWN”,  if I want to lower the water depth, so it means that I should redesign the mooring line (including changing the anchor position, changing the length of the mooring line) based on this case, so what kind of parameters can be used to quantitatively judge the feasibility of the design? The way I see it at the moment is that, after setting the new length of mooring line and anchor position, free-deacy makes the new natural  frequency consistent with the original model, then the new mooring line can be considered reasonable. I am not sure of the correctness of the method I have envisaged. So could you please give me some advice？Thanks a lot. 

Best regards,

**Reply by jjonkman** [reply]
Dear @axxyyyyyyyy378,

I agree that `EA` divided by `UnstrLen` is the axial stiffness of the mooring line "spring".

A proper mooring system design would not only result in desirable system motions and natural frequencies, but like other structural members, would also withstand ultimate and fatigue loads.

Best regards,

**Reply by axxyyyyyyyy378** [reply]
Dear @jjonkman ,

Thanks for your kind reply, now I have another idea. In the process of designing a new mooring line, I was wondering if there was a way to attach an extra weight to a certain part of the mooring line (or to one of the discrete nodes of the mooring line) ? It's like hanging an iron ingot somewhere on a mooring line. 

Best regards,

**Reply by jjonkman** [reply]
Dear @axxyyyyyyyy378,

Clump weights can certainly be modeled in MoorDyn by adding a mass (`M`) at `Connect` nodes.  See the online MoorDyn documentation for more information: https://moordyn.readthedocs.io/en/latest/inputs.html.

Best regards,

**Reply by axxyyyyyyyy378** [reply]
Dear @jjonkman ,

Thanks for your help, I have successfully added a concentrated mass to the mooring line, it seems work. With the concentrated mass as the cut-off point, the original three mooring lines, are divided into six (line1-6)(input file below) . However, there is still a problem, when running Moordyn, prompted“Catenary solve of Line 4-6 unsuccessful.” Can you help me analyze the reason? Thank you.

this is my input file “moordyn”：
<img width="553" alt="1701750851985" src="https://github.com/OpenFAST/openfast/assets/150985396/69659fab-5b99-4d96-b24c-f694355b5bf1">

This is the process of running the program given the prompt：
<img width="654" alt="1701751026715" src="https://github.com/OpenFAST/openfast/assets/150985396/6167f4cc-797b-4849-9b0d-a2bd6adf3bbc">

Best regards,

**Reply by mattEhall** [reply]
Hi, MoorDyn's catenary initial condition routine doesn't support some situations, such us when a line section is fully off the seabed. The message you're seeing indicates that this is happening. It isn't a problem. If you give it time for the dynamic relaxation stage, your initial conditions will still be the correct static equilibrium.
Matt

**Reply by axxyyyyyyyy378** [reply]
Dear @mattEhall ,

Thank you for your help, I have tried to understand it. Now I try to add a 5000kg weight on the catenary, in the wind turbine shutdown condition, without the initial displacement and external load, it is feasible, but when I try to make surge = 5, the catenary calculation on the wrong, then when I try to reduce the value of surge, such as surge = 2, the error disappeared, could you please help me to analyze the reason?

this is my catenary:
<img width="629" alt="1701839398484" src="https://github.com/OpenFAST/openfast/assets/150985396/41075791-8188-4de7-82f8-123df0eb101e">
and this is my inputfile "ELASTODYN" : **_(surge=5)_**
<img width="785" alt="1701839327130" src="https://github.com/OpenFAST/openfast/assets/150985396/617a91de-24ac-4c5c-9a49-75650929edf2">
but there is something wrong:
<img width="691" alt="1701839252494" src="https://github.com/OpenFAST/openfast/assets/150985396/d7b87f94-a7d9-4d22-8cba-9d4ae640d3ae">

when "surge=2":
![1701839750247](https://github.com/OpenFAST/openfast/assets/150985396/32c6b550-5dd7-40ba-a288-2892c0a36f62)
there is nothing wrong:
<img width="713" alt="1701839692541" src="https://github.com/OpenFAST/openfast/assets/150985396/4b674b95-8903-43b9-9fcd-3c3ef5666e20">

I have another question, how to model a buoy on the mooring line? Can i change the mass of the weight to negative? But It didn't seem to work after my test.
<img width="623" alt="1701863557288" src="https://github.com/OpenFAST/openfast/assets/150985396/fc4396ff-2c79-4205-a04e-9aad55ba9b0f">
<img width="675" alt="1701863585615" src="https://github.com/OpenFAST/openfast/assets/150985396/b82a995a-8cd6-4942-9532-192df5a03f12">

Best regards,
